### PR TITLE
[SPARK-38278][PYTHON] Add SparkContext.addArchive in PySpark

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to add `SparkContext.addArchive` in PySpark side that's added in https://github.com/apache/spark/pull/30486.

### Why are the changes needed?

To have the same API parity with the Scala side.

### Does this PR introduce _any_ user-facing change?

Yes, this PR exposes an API (`SparkContext.addArchive`) that exists in Scala side.

### How was this patch tested?

Doctest was added.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @sarutak mind taking a look when you find some time?

File: python/pyspark/context.py [hunk 1]
```diff
@@ -1278,6 +1278,53 @@ def addPyFile(self, path: str) -> None:
...
+        Read '100' as an integer and process with the data in the RDD.
+
+        >>> def func(iterator):
+        ...    with open("%s/test.txt" % SparkFiles.get("test.zip")) as f:
+        ...        v = int(f.readline())
+        ...        return [x * int(v) for x in iterator]
+        >>> sc.parallelize([1, 2, 3, 4]).mapPartitions(func).collect()
+        [100, 200, 300, 400]
+
+        Notes
```
Comment (sarutak) [inline python/pyspark/context.py:1322]: `Notes` section seems not to be allowed to appear twice.

File: python/pyspark/context.py [hunk 2]
```diff
@@ -1278,6 +1278,53 @@ def addPyFile(self, path: str) -> None:
...
+
+        .. versionadded:: 3.3.0
+
+        Notes
+        -----
+        A path can be added only once. Subsequent additions of the same path are ignored.
+
+        Examples
+        --------
+
```
Comment (sarutak) [inline python/pyspark/context.py:1300]: nit: blank line is not needed here.

**Comment by sarutak** [comment]
@HyukjinKwon Thank you for adding this feature for Python!
I think it almost LGTM except for some minor stuffs.

**Review by Yikun — APPROVED** [review]

**Review by sarutak — APPROVED** [review]
LGTM. Pending GA.

**Comment by sarutak** [comment]
Merging to `master`. Thank you @HyukjinKwon  and @Yikun !

**Comment by HyukjinKwon** [comment]
Thanks!!

**Pull request closed without merging**